.github/workflows/review.yml

```diff
@@ -43,3 +43,26 @@ jobs:
       run: mv .github/tools/.checkpatch.conf .
     - name: Run checkpatch review
       uses: webispy/checkpatch-action@v9
+  upstream_tests:
+     runs-on: self-hosted
+     timeout-minutes: 120
+     name: upstream tests
+     steps:
+     - uses: actions/checkout@v4
+       with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+     - name: 'Prepare machine'
+       run: |
+          vagrant snapshot restore clean_vm_v1
+          echo "FYI vagrant restore command finished with exit code: $?"
+          sleep 2
+          vagrant status
+          vagrant up
+          sleep 2
+     - name: 'Run tests'
+       run: |
+          vagrant ssh -c "cd host/mdadm && .github/tools/run_mdadm_tests.sh"
```

Comment:
same comment here, every step should start with "cd ..".  I do think it works without that but I think it's safest to be in the exact dir as we'll add more runners to this same sever later after you get things working with one.  They are isolated by dirs